$NetBSD: patch-ad,v 1.22 2017/06/28 15:35:48 he Exp $

*BSD and qnx are identified as linux.
Add another SDK path on Mac OS X.
All sparc cpus can not do unaligned access.
Detect NetBSD ARMv7 hardfloat toolchain.
Recognize powerpc as a target ISA, so we don't end up with generic-gnu
and possibly no libm reference...

--- build/make/configure.sh.orig	2017-01-12 20:27:27.000000000 +0000
+++ build/make/configure.sh
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/pkg/bin/bash
 ##
 ##  configure.sh
 ##
@@ -680,7 +680,7 @@ process_common_toolchain() {
       aarch64*)
         tgt_isa=arm64
         ;;
-      armv7*-hardfloat* | armv7*-gnueabihf | arm-*-gnueabihf)
+      armv7*-hardfloat* | armv7*-gnueabihf | arm-*-gnueabihf | armv7*-netbsd*-*hf)
         tgt_isa=armv7
         float_abi=hard
         ;;
@@ -697,6 +697,9 @@ process_common_toolchain() {
       *sparc*)
         tgt_isa=sparc
         ;;
+      *powerpc*)
+	tgt_isa=powerpc
+	;;
     esac
 
     # detect tgt_os
@@ -732,7 +735,7 @@ process_common_toolchain() {
         [ -z "$tgt_isa" ] && tgt_isa=x86
         tgt_os=win32
         ;;
-      *linux*|*bsd*)
+      *linux*|*bsd*|*qnx6*)
         tgt_os=linux
         ;;
       *solaris2.10)
@@ -775,6 +778,9 @@ process_common_toolchain() {
     mips*)
       enable_feature mips
       ;;
+    sparc*)
+      disable_feature fast_unaligned
+      ;;
   esac
 
   # PIC is probably what we want when building shared libs
@@ -1382,7 +1388,7 @@ EOF
   check_cc <<EOF
 unsigned int e = 'O'<<24 | '2'<<16 | 'B'<<8 | 'E';
 EOF
-    [ -f "${TMP_O}" ] && od -A n -t x1 "${TMP_O}" | tr -d '\n' |
+    [ -f "${TMP_O}" ] && od -t x1 "${TMP_O}" | tr -d '\n' |
         grep '4f *32 *42 *45' >/dev/null 2>&1 && enable_feature big_endian
 
     # Try to find which inline keywords are supported
@@ -1399,11 +1405,33 @@ EOF
         # bionic includes basic pthread functionality, obviating -lpthread.
         ;;
       *)
-        check_header pthread.h && add_extralibs -lpthread
+        check_header pthread.h && add_extralibs ${PTHREAD_LDFLAGS} ${PTHREAD_LIBS}
         ;;
     esac
   fi
 
+  case ${tgt_os} in
+  darwin*)
+     add_extralibs -lm
+     ;;
+  solaris*)
+     add_extralibs -lm -lrt
+     ;;
+  linux*)
+     case ${gcctarget} in
+     *qnx6*)
+         add_extralibs -lm
+         ;;
+     *openbsd*)
+         add_extralibs -lm
+         ;;
+     *)
+         add_extralibs -lm -lrt
+         ;;
+     esac
+     ;;
+  esac
+
   # only for MIPS platforms
   case ${toolchain} in
     mips*)
